examples/next/pages/ui/components/in-app-messaging/override-styling/index.page.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useCallback, useEffect } from 'react';
+import { Amplify } from 'aws-amplify';
+import {
+  Button,
+  InAppMessageDisplay,
+  useInAppMessaging,
+  withInAppMessaging,
+} from '@aws-amplify/ui-react';
+import '@aws-amplify/ui-react/styles.css';
+
+import config from './aws-exports';
+
+Amplify.configure(config);
+
+const CustomModalMessage = (props) => (
+  <InAppMessageDisplay.ModalMessage
+    {...props}
+    style={{ container: { backgroundColor: 'antiquewhite' } }}
+  />
+);
+
+function App() {
+  const { displayMessage } = useInAppMessaging();
+
+  const displayCustomModalMessage = useCallback(
```

Comment:
This might actually end up being less performant since `displayMessage` is already memoized and isn't very likely to change in the provider right?

I think we can just stick the `displayMessage` call right inside the `useEffect`

Reply:
Generally prefer to keep memoization out of examples, but We are reusing `displayCustomModalMessage` in the `onClick` handler of the `Button` as well, and went with memoization over repetition

Reply:
To elaborate on this, in order to use `displayCustomModalMessage` inside the `useEffect`  it needs to be memoized, else the function reference is new on each render causing the `useEffect` to run on each render